subsys/logging/log_output.c

```diff
@@ -158,6 +155,17 @@ static inline bool is_leap_year(uint32_t year)
 static void __attribute__((unused)) get_YMD_from_seconds(uint64_t seconds,
 			struct YMD_date *output_date)
 {
+#if defined(CONFIG_COMMON_LIBC_GMTIME_R) || defined(CONFIG_TC_PROVIDES_POSIX_C_LANG_SUPPORT_R)
```

Comment:
```suggestion
#if defined(CONFIG_COMMON_LIBC_GMTIME_R) || defined(CONFIG_POSIX_C_LANG_SUPPORT_R)
```

Reply:
no, we need to use `CONFIG_TC_PROVIDES_POSIX_C_LANG_SUPPORT_R`, because that is true, if the toolchain provides `gmtime_r`. `CONFIG_POSIX_C_LANG_SUPPORT_R` is just there to enable `CONFIG_COMMON_LIBC_GMTIME_R`( which will provide `gmtime_r`),  if `CONFIG_TC_PROVIDES_POSIX_C_LANG_SUPPORT_R` is not true.

Reply:
> no, we need to use `CONFIG_TC_PROVIDES_POSIX_C_LANG_SUPPORT_R`, because that is true, if the toolchain provides `gmtime_r`.

Half correct.

`CONFIG_POSIX_C_LANG_SUPPORT_R` is just there to enable `CONFIG_COMMON_LIBC_GMTIME_R`( which will provide `gmtime_r`), if `CONFIG_TC_PROVIDES_POSIX_C_LANG_SUPPORT_R` is not true.

`CONFIG_TC_PROVIDES_*` options are exclusively for negotiation between the toolchain / libc and the posix api. 

The application should still select `CONFIG_POSIX_C_LANG_SUPPORT_R` which will work the same independent of the libc.

So the application in question should select `CONFIG_POSIX_C_LANG_SUPPORT_R=y`.

Reply:
Technically, you can also use `_POSIX_C_LANG_SUPPORT_R` here as well @maas-hamburg